eth-bytecode-db/eth-bytecode-db/src/verification/db.rs

```diff
@@ -131,21 +115,112 @@ async fn insert_source_files(
                 .context("insert into \"files\"")?,
             }
         };
+        result.push(file);
+    }
+
+    Ok(result)
+}
+
+async fn insert_source_details(
+    txn: &DatabaseTransaction,
+    source: types::Source,
+    file_models: &[files::Model],
+) -> Result<(sources::Model, bool), anyhow::Error> {
+    let abi = source
+        .abi
+        .map(|abi| serde_json::from_str(&abi).context("deserialize abi"))
+        .transpose()?;
+
+    let file_ids: BTreeSet<_> = file_models.iter().map(|file| file.id).collect();
 
+    // Results in `SELECT md5({1,2,3}::text)
+    let file_ids_hash_query = Statement::from_sql_and_values(
+        DatabaseBackend::Postgres,
+        "SELECT md5($1::text)::uuid",
+        [sea_orm::Value::Array(
+            ArrayType::BigInt,
```

Comment:
can't you make `file_ids.into_iter().map(..).collect::<Vec<_>>().into()`?
https://docs.rs/sea-orm/latest/sea_orm/enum.Value.html#impl-From%3CVec%3CT%2C%20Global%3E%3E-for-Value